src/Featurizers/Components/UnitTests/FilterFeaturizerImpl_UnitTest.cpp

```diff
@@ -0,0 +1,552 @@
+// ----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#define CATCH_CONFIG_MAIN
+#include "catch.hpp"
+
+#include "../FilterFeaturizerImpl.h"
+
+#include "../../TestHelpers.h"
+#include "../../../Archive.h"
+
+namespace NS                                = Microsoft::Featurizer;
+namespace Components                        = Microsoft::Featurizer::Featurizers::Components;
+
+class SimpleSumTransformer : public NS::StandardTransformer<int, int> {
+public:
+    // ----------------------------------------------------------------------
+    // |  Public Types
+    using BaseType                          = NS::StandardTransformer<int, int>;
+
+    // ----------------------------------------------------------------------
+    // |  Public Methods
+    SimpleSumTransformer(int delta=0) : _delta(delta) {}
+    SimpleSumTransformer(Archive &ar) : SimpleSumTransformer(NS::Traits<int>::deserialize(ar)) {}
+    ~SimpleSumTransformer(void) override = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(SimpleSumTransformer);
+
+    bool operator==(SimpleSumTransformer const &other) const {
+        return _delta == other._delta;
+    }
+
+    bool operator!=(SimpleSumTransformer const &other) const {
+        return (*this == other) == false;
+    }
+
+    void save(Archive &ar) const override {
+        NS::Traits<int>::serialize(ar, _delta);
+    }
+
+private:
+    // ----------------------------------------------------------------------
+    // |  Private Data
+    int const                               _delta;
+
+    // ----------------------------------------------------------------------
+    // |  Private Methods
+    void execute_impl(typename BaseType::InputType const &value, typename BaseType::CallbackFunction const &callback) override {
+        callback(value + _delta);
+    }
+};
+
+TEST_CASE("SimpleSumTransformer") {
+    // ----------------------------------------------------------------------
+    using FilterTransformer =
+        Components::FilterTransformer<
+            SimpleSumTransformer,
+            std::tuple<char, int>,

```

Comment:
Does it make a difference that these aren't const refs?

Reply:
I realized this as I was writing tests and got lazy. I think that I should add code to verify that all input elements are const references and update tests, but I think that I will do that as a separate PR.